contracts/signature-validators/OffChainSignatureValidator.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.20;
+
+import { IERC1271 } from "openzeppelin/contracts/interfaces/IERC1271.sol";
+import { Strings } from "openzeppelin/contracts/utils/Strings.sol";
+import { Party } from "../party/Party.sol";
+
+/// @notice Contract that by default validates off-chain signature for parties
+contract OffChainSignatureValidator is IERC1271 {
+    error NotMemberOfParty();
+    error InsufficientVotingPower();
+    error MessageHashMismatch();
+
+    event SigningThresholdBipsSet(
+        Party indexed party,
+        uint96 oldThresholdBips,
+        uint96 newThresholdBips
+    );
+
+    mapping(Party party => uint96 thresholdBips) public signingThersholdBips;
+
+    /// @notice Validate an off-chain signature
+    /// @dev This function requires `signature` to be a valid EOA signature from a member in the
+    /// party with sufficient voting power. The raw message must be abi encoded and appended to
+    /// the end of the signature. EIP-712 typed signatures are not supported.
+    function isValidSignature(bytes32 hash, bytes memory signature) external view returns (bytes4) {
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+        assembly {
+            // First word of signature after size contains r
+            r := mload(add(signature, 0x20))
+            s := mload(add(signature, 0x40))
+            // v is one byte which starts after s. type is uint8 so extra data will be ignored
+            v := mload(add(signature, 0x41))
+        }
+
+        bytes memory message;
```

Comment:
I expected there would be some kind of check on the message, e.g. ensuring that it is a message to sign-in to an app. What are we doing with `message`?

Reply:
Our conclusion so far was to allow all "plain text" signatures as it would be impossible to actually determine what's happening for all the various SIWE like schemas. The hash matching requirment enforces this. See lines 45-53

Reply:
I see. This is to ensure it is a plain text signed message because that is what is generally used by SIWE? And ensuring it is not something else like, for example, signed typed data?

Reply:
Yes exactly. Generally signatures that deal with contract level execution and tokens are ERC712 typed signatures which are formatted differently and therefore not supported by this implementation (by design).